pkg/metrics/metrics.go

```diff
@@ -183,3 +185,90 @@ func (r *standardRegistry) ServiceRetriesCounter() metrics.Counter {
 func (r *standardRegistry) ServiceServerUpGauge() metrics.Gauge {
 	return r.serviceServerUpGauge
 }
+
+// ScalableHistogram is a Histogram with a predefined time unit used when
+// producing observations allowing the call to Start() and ObserveDuration()
+// without explicitly setting the observed value.
+type ScalableHistogram interface {
+	With(labelValues ...string) ScalableHistogram
+	Start()
+	Observe(v float64)
+	ObserveDuration()
+}
+
+// HistogramWithScale is a histogram that will convert its observed value to the specified unit.
+type HistogramWithScale struct {
+	histogram metrics.Histogram
+	unit      time.Duration
+	timer     *metrics.Timer
+}
+
+// With implements ScalableHistogram.
+func (s *HistogramWithScale) With(labelValues ...string) ScalableHistogram {
+	s.histogram = s.histogram.With(labelValues...)
+	return s
+}
+
+// Start implements ScalableHistogram.
+func (s *HistogramWithScale) Start() {
+	s.timer = metrics.NewTimer(s.histogram)
+	s.timer.Unit(s.unit)
+}
```

Comment:
I'm not sure we need any sort of pseudo-atomicity for the multi-histogram, but if we do, 
we could have a private method, such as:

func (s *HistogramWithScale) start(t time.Time)

which instantiates a Timer with that specific time.
then Start would call s.start(time.Now())
and the multi histogram would do:
t := time.Now()
and call s.start(t) for each of the histograms in the range.

Reply:
:+1: done